tests/integration/test_app_create.py

```diff
@@ -0,0 +1,165 @@
+import unittest
+from uuid import UUID
+from mock import patch
+
+from flask import Flask
+from flask_babel import Babel
+import app
+from app import settings
+
+class TestCreateApp(unittest.TestCase):
+    def setUp(self):
+        pass
+
+    def test_returns_application(self):
+        self.assertIsInstance(app.create_app(), Flask)
+
+    # Should new relic have direct access to settings?
+    # Probably better for create app to have the control
+    def test_setups_newrelic(self):
+        with patch('newrelic.agent.initialize') as new_relic:
+            settings.EQ_NEW_RELIC_ENABLED = "True"
+            app.create_app()
+            self.assertEqual(new_relic.call_count, 1)
+
+    def test_sets_static_url(self):
+        self.assertEqual('/s', app.create_app().static_url_path)
+
+    def test_sets_static_folder_that_exists(self):
+        self.assertRegex(app.create_app().static_folder, '../static$')
+
+    def test_sets_content_length(self):
+        self.assertGreater(app.create_app().config['MAX_CONTENT_LENGTH'], 0)
+
+    def test_enforces_secure_session(self):
+        application = app.create_app()
+        self.assertTrue(application.secret_key)
+        self.assertTrue(application.permanent_session_lifetime)
+        self.assertTrue(application.session_interface)
+        # All tests run in dev mode which sets SESSION_COOKIE_SECURE to false
+        self.assertFalse(application.config['SESSION_COOKIE_SECURE'])
+
+    # localisation may not be used but is currently attached...
+    def test_adds_i18n_to_application(self):
+        self.assertIsInstance(app.create_app().babel, Babel) # pylint: disable=no-member
+
+    def test_adds_logging_of_request_ids(self):
+        with patch('app.logger') as logger:
+            settings.EQ_DEV_MODE = True
+            settings.EQ_APPLICATION_VERSION = False
+            application = app.create_app()
+
+            application.test_client().get('/')
+            self.assertEqual(1, logger.new.call_count)
+            _, kwargs = logger.new.call_args
+            self.assertTrue(UUID(kwargs['request_id'], version=4))
+
+
+    def test_enforces_secure_headers(self):
+        client = app.create_app().test_client()
+        headers = client.get('/').headers
+        self.assertEqual('no-cache, no-store, must-revalidate', headers['Cache-Control'])
+        self.assertEqual('no-cache', headers['Pragma'])
+        self.assertEqual('max-age=31536000; includeSubdomains', headers['Strict-Transport-Security'])
+        self.assertEqual('DENY', headers['X-Frame-Options'])
+        self.assertEqual('1; mode=block', headers['X-Xss-Protection'])
+        self.assertEqual('nosniff', headers['X-Content-Type-Options'])
+
+
+    # Indirectly covered by higher level integration
+    # tests, keeping to highlight that create_app is where
+    # it happens.
+    def test_adds_blueprints(self):
+        self.assertGreater(len(app.create_app().blueprints), 0)
+
+    # covered by session/test_login.py
```

Comment:
I left them in because I thought they *might* be useful to anyone who comes after who might wonder why functionality X of create_app() isn't covered in this file.

I'm happy to remove them though, or I could remove the defs and just keep one line comments highlighting where the behaviour is tested?